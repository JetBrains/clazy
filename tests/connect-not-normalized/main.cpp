#include <QtCore/QObject>
#include <QtCore/QVariant>

void test()
{
    QObject o;

    QVariant v, returnedValue;
    QMetaObject::invokeMethod(&o, "mySlot",
                              Q_RETURN_ARG(QVariant, returnedValue), // OK
                              Q_ARG(const QVariant, v)); // Warn
    QMetaObject::invokeMethod(&o, "mySlot",
                              Q_RETURN_ARG(QVariant, returnedValue), // OK
                              Q_ARG(const QVariant, // Warn
                                    v)); // multi-line

    Q_ARG(QVariant , v); // OK
    Q_ARG(QVariant   &, v); // Warn
    Q_ARG(QVariant&, v); // OK
    Q_ARG(const QVariant &, v); // Warn
}

void testConnect()
{
    QObject o;
    o.connect(&o, SIGNAL(destroyed(int, int)), // Warn
              &o, SLOT(void foo(const int))); // Warn
    o.connect(&o, SIGNAL(destroyed(int,int)), // OK
              &o, SLOT(void foo(int))); // OK
}
