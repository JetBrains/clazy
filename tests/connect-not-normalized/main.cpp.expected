connect-not-normalized/main.cpp:11:31: warning: Signature is not normalized. Use QVariant instead of const QVariant [-Wclazy-connect-not-normalized]
connect-not-normalized/main.cpp:14:31: warning: Signature is not normalized. Use QVariant instead of const QVariant [-Wclazy-connect-not-normalized]
connect-not-normalized/main.cpp:18:5: warning: Signature is not normalized. Use QVariant& instead of QVariant & [-Wclazy-connect-not-normalized]
connect-not-normalized/main.cpp:20:5: warning: Signature is not normalized. Use QVariant instead of const QVariant & [-Wclazy-connect-not-normalized]
connect-not-normalized/main.cpp:26:19: warning: Signature is not normalized. Use destroyed(int,int) instead of destroyed(int, int)
connect-not-normalized/main.cpp:27:19: warning: Signature is not normalized. Use void foo(int) instead of void foo(const int)
